content/docs/styled-system/color-mode.mdx

```diff
@@ -80,36 +90,75 @@ export default theme
 > Remember to pass your custom `theme` to the `ChakraProvider`, otherwise your
 > color mode config won't be taken into consideration.
 
-#### Behavior of ColorMode
+#### Common Configurations
 
-The current hierarchy of how the color mode is defined is as follows:
+To help you define you desired behavior, here we define all common usage of the theme config :
 
-- if `useSystemColorMode` is true `system`-color will be used as default -
-  `initialColorMode` is the fallback if system color mode can't be resolved
+> **Tip** If you prefer, you can use this [playground](https://codesandbox.io/s/chakra-ui-color-mode-test-f5fcwr?file=/src/uiLayer.chakra/customTheme.ts) to try different values.
 
-- if `data-theme` prop is defined through e.g. `ColorModeScript` / after
-  modification/initial load of the colorMode this value will be used
+> **Note** The hook `useColorMode` let you update color mode of your app in every cases. [Learn more](#changing-color-mode)
+
+```tsx live=false
+// A.
+// System sets initial value.
+// App subscribes to system color mode changes.
+const config: ThemeConfig = {
+  initialColorMode: 'system',
+  useSystemColorMode: true,
+}
+
+// B.
+// System sets initial value.
+// App color mode is detached from system color mode changes.
+const config: ThemeConfig = {
+  initialColorMode: 'system',
+  useSystemColorMode: false,
+}
+
+// C.
+// You choose initial value.
+// App color mode is detached from system color mode changes.
+const config: ThemeConfig = {
+  initialColorMode: 'dark', // 'dark' | 'light'
+  useSystemColorMode: false,
+}
+
+
+// D.
+// You choose initial value.
+// App subscribes to system color mode changes.
+const config: ThemeConfig = {
+  initialColorMode: 'dark', // 'dark' | 'light'
+  useSystemColorMode: true,
+}
+```
+
+#### Behavior of ColorMode
+
+The current hierarchy of how chakra resolves the color mode:
 
-- if `colorModeManager` = `localStorage` and a value is defined for
-  `chakra-ui-color-mode` this will be used
+- Get the color mode value in the specified storage (localStorage or cookie manager or custom)
 
-- if `initialColorMode` = `system` system-color will be used as default -
-  `initialColorMode` is the fallback if system color mode isn't resolved
+- If value doesn't exist, use the `initialColorMode` value specified.
+  - If the initial value is `system`, then we'll compute the color mode using
+    `matchMedia`
+  - Else, we use the initial value as is.
 
-- if `initialColorMode` = `'light'|'dark'` the corresponding value will be used
+- If user specifies `useSystemColorMode: true`, then we'll subscribe to color
+  mode changes from the operating system. It is no longer used to determine the
+  initial color mode. To achieve that, please use `initialColorMode: "system"`
 
-We currently accept 3 different values for `initialColorMode`:
-`light`,`dark`,`system`
+// TODO : Keep this ??? Is still valid ??
+- if `data-theme` prop is defined through e.g. `ColorModeScript` / after
```

Comment:
**Need Help**
This sentence is still valid ??
> if `data-theme` prop is defined through e.g. `ColorModeScript` / after modification/initial load of the colorMode this value will be used

Reply:
I'm not sure I understand what you mean by this sentence.

Reply:
I'm not aware if this sentence is still valid, because i take a too short look at the code to be able to deduce it :)

Sentence i'm referring to:
> if data-theme prop is defined through e.g. ColorModeScript / after modification/initial load of the colorMode this value will be used
```

Reply:
After commit 1d1ec8bb6a4a658357504dbde6b8077c8ac6cfb0 this is not already resolved.

Reply:
I'm testing the `data-theme` attribute on html element in this [Codesandbox](https://codesandbox.io/s/chakra-ui-color-mode-test-html-attr-data-theme-l2sinw?file=/src/App.tsx).

But i'm not sure if  this "feature" should be included in the docs, and how to explain it.

My doubts:
- How this is supposed to work? 
- Why developer want to mutate this attribute ?
  - Why toggle this attribute instead of using `useColorMode.toggleColorMode` ?
  - Maybe to toggle color mode outside React ?

Reply:
We should consider to discard these lines from the docs because of lack of explained usage ?
Or we should keep these lines as they are ?

Reply:
Hey, sorry on the inactivity on my end. You are right, I don't think this would be needed in the docs since it doesn't provide any useful information and it causes confusion.

Reply:
Don't worry, we are all busy.
I removed this sentence.